src/sagas/search.js

```diff
@@ -0,0 +1,41 @@
+import { call, takeEvery, put, select } from 'redux-saga/effects';
+import search from '../actions/search';
+import { getSearch, getGif} from '../services/api'
+
+
+export function* loadSearchGifs(action) {
+  try {
+    const searchResponse = yield call(getSearch, action.payload);
+    yield put(search.creators.requestSearchGifsSuccess(searchResponse.data));
+  } catch (error) {
+    yield put(search.creators.requestSearchGifsFail(error));
+    console.log(error);
+  }
+}
+
+export function* loadSearchFavorites(action) {
+  try {
+    const selector = yield select()
+    const favs = selector.favoriteGifs
+    const regex = RegExp(`.*(${action.payload})+.*`,'gi')
+    const response = favs.filter(fav => regex.test(fav.title))
+    yield put(search.creators.searchFavsSuccess(response))
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+export function* loadSearchGif(action) {
```

Comment:
This function and variable names are very similar to the one from the previous saga `loadSearchGifs` but this saga is actually for getting the details of a Gif, so I would suggest changing the names to reflect that.